packages/valory/skills/twitter_scoring_abci/rounds.py

```diff
@@ -77,7 +85,20 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
             if self.most_voted_payload == json.dumps(
                 self.ERROR_PAYLOAD, sort_keys=True
             ):
-                return self.synchronized_data, Event.API_ERROR
+                api_retries = (
+                    cast(SynchronizedData, self.synchronized_data).api_retries + 1
+                )
+
+                if api_retries >= MAX_API_RETRIES:
+                    return self.synchronized_data, Event.DONE
+
+                synchronized_data = self.synchronized_data.update(
+                    synchronized_data_class=SynchronizedData,
+                    **{
+                        get_name(SynchronizedData.api_retries): api_retries,
+                    }
+                )
+                return synchronized_data, Event.API_ERROR
```

Comment:
This logic should be on the behaviour side. In essence, `API_ERROR` event should trigger the same round, then the behaviour would check the `api_retries`, retry, update the `api_retries`, and pass them to the payload. If `api_retries >= MAX_API_RETRIES` the behaviour would return another kind of `ERROR_PAYLOAD`, e.g., `RETRIES_EXCEEDED_PAYLOAD`. The round would detect it and return an `Event.DONE`.

The round would only update the `api_retries` given by the payload and perform no logic based on it.

The current implementation gets the work done and is not a priority to be refactored at the moment, therefore let's merge as is, but I would suggest opening an issue for the future as it is important to follow the best practices, i.e., no logic on the round side.